code/datums/datum.dm

```diff
@@ -5,7 +5,7 @@
 	/// Status traits attached to this datum
 	var/list/status_traits
 	var/list/comp_lookup
-	var/list/signal_procs
+	var/list/list/datum/callback/signal_procs
```

Comment:
but why thoughhhh

Reply:
Issue was:
```
code/datums/components/_component.dm, line 151, column 26:
warning: field access requires static type: "len"
- code/datums/datum.dm:8:24: add a type annotation after /list here
```
_component.dm, L151 is `if(!signal_procs[target].len)`, and theres no major indication that `signal_procs[target]` is a list, so its double-listed for a reason.

I used https://github.com/tgstation/tgstation/pull/43087/files as a baseline on how to fix this, and this https://github.com/tgstation/tgstation/pull/43087/files#r266247719 comment explains it well.